client/src/Dataset/List/list.hooks.ts

```diff
@@ -1,58 +1,149 @@
-import { useState, useEffect, Dispatch, SetStateAction } from 'react';
+import { useState, useEffect, Dispatch, SetStateAction, useMemo } from 'react';
 import { useNavigation } from 'react-navi';
-import { FormikConfig, FormikHelpers } from 'formik';
-import * as Yup from 'yup';
 
-import { Dataset } from '../../common/types';
+import {
+    Dataset,
+    Category,
+    UserInfo,
+    DatasetPermissions,
+    IDict,
+} from '../../common/types';
 import * as DatasetApi from '../datasets.api';
+import * as AdminPanelApi from '../../AdminPanel/adminPanel.api';
 import useGlobalContext from '../../common/hooks/useGlobalContext';
+import useAuthContext from '../../common/hooks/useAuthContext';
 import { addProcess, removeProcess } from '../../common/utils/globalActions';
 
 // interfaces
 interface DatasetsState {
     list: ListState;
     dialogs: DialogsState;
+    edit: [
+        DatasetWithCategories | null,
+        Dispatch<SetStateAction<DatasetWithCategories | null>>,
+    ];
+    share: [
+        DatasetWithCategories | null,
+        Dispatch<SetStateAction<DatasetWithCategories | null>>,
+    ];
     navigation: NavigationState;
+    datasetWithCategories: DatasetWithCategories[];
+    onDeleteClick(id: number): Promise<void>;
+    onCocoDownloadClick(name: string, id: number): Promise<void>;
 }
 interface NavigationState {
-    openDetails(dataset: Dataset): void;
+    openDetails(dataset: DatasetWithCategories): void;
 }
 interface ListState {
-    page: number;
+    offset: [number, Dispatch<SetStateAction<number>>];
     pageCount: number;
     datasets: Dataset[];
-    categories: string[];
+    categories: Category[];
+    tags: string[];
+    usernames: string[];
     setPage: Dispatch<SetStateAction<number>>;
     refreshPage(): void;
 }
 interface DialogsState {
     help: [boolean, Dispatch<SetStateAction<boolean>>];
     create: [boolean, Dispatch<SetStateAction<boolean>>];
-    share: [boolean, Dispatch<SetStateAction<boolean>>];
-    edit: [boolean, Dispatch<SetStateAction<boolean>>];
 }
-interface FormCreateState {
+export interface DatasetWithCategories {
+    id: number;
     name: string;
-    categories: Array<string>;
+    directory: string;
+    first_image_id?: number;
+    numberAnnotated: number;
+    numberImages: number;
+    categories: Category[];
+    owner: string;
+    users: Array<string>;
+    annotate_url: string;
+    default_annotation_metadata: IDict;
+    deleted: boolean;
+    deleted_date?: {
+        $data: number;
+    };
+    permissions?: DatasetPermissions;
 }
 
+const downloadURI = (uri: string, exportName: string) => {
```

Comment:
this function doesn't work with the react hook api so it should be moved to some kind of `utils` module